libraries/AP_WindVane/AP_WindVane.cpp

```diff
@@ -0,0 +1,320 @@
+    AP_GROUPINFO("CAL", 8, AP_WindVane, _calibration, 0),
+
+    // @Param: ANA_DZ
+    // @DisplayName: Analog potentiometer dead zone
+    // @Description: Analog potentiometer mechanical dead zone
+    // @Units: deg
+    // @Increment: 1
+    // @Range: 0 360
+    // @User: Standard
+    AP_GROUPINFO("ANA_DZ", 9, AP_WindVane, _analog_deadzone, 0),
+
+    AP_GROUPEND
+};
+
+// create a global instances of low pass filter
+LowPassFilterFloat low_pass_filter_wind_sin = LowPassFilterFloat(2.0f);
+LowPassFilterFloat low_pass_filter_wind_cos = LowPassFilterFloat(2.0f);
+LowPassFilterFloat low_pass_filter_wind_speed = LowPassFilterFloat(2.0f);
+
+// Public
+// ------
+
+// constructor
+AP_WindVane::AP_WindVane()
+{
+    AP_Param::setup_object_defaults(this, var_info);
+    if (_s_instance) {
+        AP_HAL::panic("Too many Wind Vane sensors");
+    }
+    _s_instance = this;
+}
+
+// destructor
+AP_WindVane::~AP_WindVane(void)
+{
+}
+
+/*
+ * Get the AP_WindVane singleton
+ */
+AP_WindVane *AP_WindVane::get_instance()
+{
+    return _s_instance;
+}
+
+// return true if wind vane is enabled
+bool AP_WindVane::enabled() const
+{
+    return (_type != WINDVANE_NONE);
+}
+
+// Initialize the Wind Vane object and prepare it for use
+void AP_WindVane::init()
+{
+    // a pin for reading the Wind Vane voltage
+    windvane_analog_source = hal.analogin->channel(ANALOG_INPUT_NONE);
+}
+
+// Update wind vane, called at 10hz
+void AP_WindVane::update()
+{
+    // exit immediately if not enabled
+    if (!enabled()) {
+        return;
+    }
+
+    // check for calibration
+    calibrate();
+
+    update_apparent_wind_direction();
+}
+
+// record home heading for use as wind direction if no sensor is fitted
+void AP_WindVane::record_home_headng()
+{
+    _home_heading = AP::ahrs().yaw;
+}
+
+// Private
+// -------
+
+// read the Wind Vane value from an analog pin, calculate bearing from analog voltage
+// assumes voltage increases as wind vane moves clockwise
+float AP_WindVane::read_analog()
+{
+    windvane_analog_source->set_pin(_analog_pin_no);
+    _current_analog_voltage = windvane_analog_source->voltage_average();
+
+    float current_analog_voltage_constrain = constrain_float(_current_analog_voltage,_analog_volt_min,_analog_volt_max);
+    float voltage_ratio = linear_interpolate(0.0f, 1.0f, current_analog_voltage_constrain, _analog_volt_min, _analog_volt_max);
+
+    float bearing = (voltage_ratio * radians(360 - _analog_deadzone)) + radians(_analog_head_bearing_offset);
+
+    return wrap_PI(bearing);
+}
+
+// read the bearing value from a PWM value on a RC channel (+- 45deg)
+float AP_WindVane::read_PWM_bearing()
+{
+    RC_Channel *ch = rc().channel(_rc_in_no-1);
+    if (ch == nullptr) {
+        return 0.0f;
+    }
+    float bearing = ch->norm_input() * radians(45);
+
+    return wrap_PI(bearing);
+}
+
+// read the apparent wind direction in radians from SITL
+float AP_WindVane::read_direction_SITL()
+{
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
```

Comment:
With the `#if` check at the caller you can just surround this entire implementation of the `read_direction_STIL` in the SITL check. (Results in slightly smaller builds, and easier to read the actual body of the function).

Reply:
great, moved the #if to surround the function's implementation.  I've also added it around the declaration in the .h file.  This isn't strictly necessary I think and makes it slightly messy but..